### PR TITLE
Refactor: simplify setting of useArrayBounds

### PR DESCRIPTION
**Pull request by WalterBright**
1. Add enum to convey meaning.
2. Straighten out overly complex logic.
3. Alphabetize commands in help message.

Refactor only - should not be any change in functionality.

### CODE REVIEW

File: src/globals.h [hunk 1]
```diff
@@ -57,9 +57,7 @@ struct Param
     bool useInvariants; // generate class invariant checks
     bool useIn;         // generate precondition checks
     bool useOut;        // generate postcondition checks
-    char useArrayBounds; // 0: no array bounds checks
-                         // 1: array bounds checks for safe functions only
-                         // 2: array bounds checks for all functions
+    char useArrayBounds; // enum BOUNDSCHECK
```
Comment (yebblies) [inline src/globals.h]: Should use BOUNDSCHECK as the type

Comment (WalterBright) [inline src/globals.h]: ok

**Comment by eco** [comment]
Good. When I add boundcheck I hadn't really changed anything serious with dmd before so I tried to not rock the boat too much and made it very similar to what was already there, though I was cognizant  of how ugly it was when I added it (I believe @AndrejMitrovic pointed it out during the review too).

This is much better.

**Comment by WalterBright** [comment]
Making things flow better is always an ongoing learning process, for myself included. I generally hate all the code I wrote more than a couple years back.

**Comment by yebblies** [comment]
Auto-merge toggled on

**Pull request merged**